includes/managers/icons/migrations.php

```diff
@@ -0,0 +1,307 @@
+<?php
+namespace Elementor\Icons_Manager;
+
+use Elementor\Icons_Manager;
+use Elementor\Utils;
+use Elementor\Plugin;
+
+use Elementor\Core\Common\Modules\Ajax\Module as Ajax;
+
+use Elementor\Core\Upgrade\Manager as Upgrade_Manager;
+use Elementor\Core\Upgrade\Upgrade_Utils;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Migrations {
+
+	const NEEDS_UPDATE_OPTION = 'icon_manager_needs_update';
+
+	const FONTAWESOME_CURRENTLY_USED_VERSION_OPTION = 'fontawesome_current_version';
+
+	/**
+	 * The latest version that requires migration.
+	 * This is needed as not every FA version update will require migration.
+	 * Migration is required only when icons are removed or renamed in the newer version.
+	 * If the version defined here is higher than the current FA version used on the website, migration will be required.
+	 */
+	const LATEST_MIGRATION_REQUIRED_VERSION = '6.5.1';
+
+	public static function get_needs_upgrade_option() {
+		return get_option( 'elementor_' . self::NEEDS_UPDATE_OPTION, null );
+	}
+
+	/**
+	 * Font Awesome icon values migration between versions
+	 *
+	 * used to convert string value of Icon control to array value of Icons control
+	 * ex: 'fa fa-star' => [ 'value' => 'fas fa-star', 'library' => 'fa-solid' ]
+	 *
+	 * @param $value
+	 *
+	 * @return array
+	 */
+	public static function fa_icon_value_migration( $value ) {
+		if ( '' === $value ) {
+			return [
+				'value' => '',
+				'library' => '',
+			];
+		}
+
+		if ( is_string( $value ) ) {
+			$value = [
+				'value' => $value,
+				'library' => '',
+			];
+		}
+
+		foreach ( self::get_migration_dictionary() as $mapping_by_version ) {
+			if ( isset( $mapping_by_version[ $value['value'] ] ) ) {
+				$value = $mapping_by_version[ $value['value'] ];
+			}
+		}
+
+		// Make sure there's no value in the old 'fa ' format, even if no replacement found.
+		if ( '' === $value['library'] ) {
+			$value = [
+				'value'   => 'fas ' . str_replace( 'fa ', '', $value ),
+				'library' => 'fa-solid',
+			];
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Get the migration dictionary (mapping of the icon changes) by versions.
+	 *
+	 * @return array
+	 */
+	public static function get_migration_dictionary() {
```

Comment:
You only using it here, is it need to be public?

Reply:
Good catch, changed.